File: public/usage-examples/graphics/draw_circle-2-animation.py
from splashkit import *

window = open_window("Circle Animation", 800, 600)

x = 0

while not quit_requested():
    process_events()
    clear_screen_to_white()

    draw_circle(color_red(), x, 300, 50)

    x += 2
    if x > 800:
        x = 0

    refresh_screen_with_target_fps(60)

File: public/usage-examples/graphics/draw_circle-2-animation.cpp
#include "splashkit.h"

using namespace splashkit;

int main()
{
    open_window("Circle Animation", 800, 600);

    double x = 0;

    while (!window_close_requested("Circle Animation"))
    {
        process_events();
        clear_screen(COLOR_WHITE);

        draw_circle(COLOR_RED, x, 300, 50);

        x += 2;
        if (x > 800) x = 0;

        refresh_screen(60);
    }

    return 0;
}